api_tests/lib/bitcoind_instance.ts

```diff
@@ -0,0 +1,77 @@
+import { ChildProcess, spawn } from "child_process";
+import * as fs from "fs";
+import tmp from "tmp";
+import { promisify } from "util";
+import { sleep } from "./util";
+
+const openAsync = promisify(fs.open);
+
+export class BitcoindInstance {
+    private process: ChildProcess;
+    private dbDir: any;
+    private username: string;
+    private password: string;
+
+    constructor(
+        private readonly projectRoot: string,
+        private readonly logDir: string
+    ) {}
+
+    public async start() {
+        const bin = process.env.BITCOIND_BIN
+            ? process.env.BITCOIND_BIN
+            : this.projectRoot +
+              "/blockchain_nodes/bitcoin/bitcoin-0.17.0/bin/bitcoind";
+
+        this.dbDir = tmp.dirSync();
+
+        this.process = spawn(
+            bin,
+            [
+                `-datadir=${this.dbDir.name}`,
+                "-regtest",
+                "-server",
+                "-printtoconsole",
+                "-bind=0.0.0.0:18444",
```

Comment:
:mega: I'd be in favour of maintaining our current functionality and choose a free port here instead of assuming that 18444 will be available.

Reply:
Good idea. How would we do that though? :)

Reply:
I would use something like this: https://www.npmjs.com/package/get-port